main.py

```diff
@@ -68,14 +83,88 @@ def printBaseEncoding(encodedVersion: List[List[Union[SudokuNumber, EmptyCell]]]
     for cell in base_encoding:
         print(" ".join(map(str, cell)) + " 0")
 
+    printNoDoubleRowEncoding(base_encoding, encodedVersion, max_value)
+        
+
+def printBaseEncoding(encodedVersion: List[List[Union[SudokuNumber, EmptyCell]]]) -> None:
+    """Base encoding for every cell contains at least one number"""
+    # Iterator used to generate a new number every time next is called
+    # on it.    
+    iter = varGenerator()
+    possible_values = 9
+    # May have over done it on the list composition here
+    base_encoding: list[list[int]] = [
+        [next(iter) for _ in range(possible_values)]
+        for row in encodedVersion
+        for cell in row
+    ]
+
+    max_value = next(iter) - 1
+    possible_values = 9
+    print("c Every cell contains at least one number")
+    print(f"p cnf {max_value} {max_value//possible_values}")
+    for cell in base_encoding:
+        print(" ".join(map(str, cell)) + " 0")
+
+
+def printNoDoubleRowEncoding(listOfInputs: List[List[int]], puzzleList: List[List[Union[SudokuNumber, EmptyCell]]], maxValues: int) -> None:
+    i = -1
+    for puzzleRow in puzzleList:
+        #initialize list of values that may not be repeated
+        forbiddenValues = []
+        i = i + 1
+        for puzzleCell in puzzleRow:
+            if isinstance(puzzleCell, SudokuNumber):
+                ##This doesn't seem to work correctly.
+                ##It SHOULD set all non-matching literals in a given list to negative if the value is set
+                print("This cell has a value!")
+                for literal in listOfInputs[i]:
+                    print(baseNineSingleVal(literal), puzzleCell.number_string)
+                    if baseNineSingleVal(literal) != puzzleCell.number_string:
+                        literal = -literal
+                        forbiddenValues.append(puzzleCell.number_string)
+            else:
```

Comment:
I think you meant to indent this else one more level. Since right now literal is not going to be bound to anything